File: 3_xmlrpc/if4030/xmlrpc/RandomServer.py

import random

from xmlrpc.server import SimpleXMLRPCServer
from xmlrpc.server import SimpleXMLRPCRequestHandler

# Restrict to a particular path.
class RequestHandler( SimpleXMLRPCRequestHandler ):
    rpc_paths = ( '/IF4030', )

class RandomServer:
    def __init__(self):
        self.min = 0
        self.max = 10
    
    def setBounds(self, min, max):
        self.min = min
        self.max = max
        return 0
    
    def getRandom(self):
        return int(random.random() * (self.max - self.min + 1)) + self.min

# Create server
with SimpleXMLRPCServer(( 'localhost', 8000 ),
                        requestHandler = RequestHandler ) as server:

    server.register_instance( RandomServer() )
    print( "server ready" )

    server.serve_forever()

File: 2_rmi/if4030/rmi/RandomServer.java
package if4030.rmi;
        
import java.rmi.registry.Registry;
import java.rmi.registry.LocateRegistry;
import java.rmi.RemoteException;
import java.rmi.server.UnicastRemoteObject;
        
public class RandomServer implements IRandom {
    private int min = 0;
    private int max = 10;
        
    public void setBounds( int min, int max ) throws RemoteException {
        this.min = min;
        this.max = max;
    }
    
    public int nextRandom() throws RemoteException {
        return (int)Math.floor(Math.random() * (max - min + 1)) + min;
    }

    public static void main( String args[] ) {
        System.setProperty( "java.rmi.server.hostname", "localhost" );
        try {
            RandomServer random = new RandomServer();
            IRandom stub = ( IRandom ) UnicastRemoteObject.exportObject( random, 0 );
            
            LocateRegistry.createRegistry( 10099 );
            Registry registry = LocateRegistry.getRegistry( "localhost", 10099 );
            registry.bind( "Random", stub );

            System.err.println( "Server ready" );
        }
        catch( Exception e ) {
            System.err.println( "Server exception: " + e.toString() );
        }
    }
}
